pilot/pkg/bootstrap/server.go

```diff
@@ -1002,21 +1002,21 @@ func (s *Server) initControllers(args *PilotArgs) error {
 // initNamespaceController initializes namespace controller to sync config map.
 func (s *Server) initNamespaceController(args *PilotArgs) {
 	if s.CA != nil && s.kubeClient != nil {
+		// create namespace controller
+		nsController := kubecontroller.NewNamespaceController(s.fetchCARoot, s.kubeClient)
 		s.addTerminatingStartFunc(func(stop <-chan struct{}) error {
-			leaderelection.
-				NewLeaderElection(args.Namespace, args.PodName, leaderelection.NamespaceController, s.kubeClient.Kube()).
-				AddRunFunction(func(leaderStop <-chan struct{}) {
-					log.Infof("Starting namespace controller")
-					nc := kubecontroller.NewNamespaceController(s.fetchCARoot, s.kubeClient)
-					// Start informers again. This fixes the case where informers for namespace do not start,
-					// as we create them only after acquiring the leader lock
-					// Note: stop here should be the overall pilot stop, NOT the leader election stop. We are
-					// basically lazy loading the informer, if we stop it when we lose the lock we will never
-					// recreate it again.
-					s.kubeClient.RunAndWait(stop)
-					nc.Run(leaderStop)
-				}).
-				Run(stop)
+			// Start informers again. This fixes the case where informers for namespace do not start,
```

Comment:
I never fully understood this comment - and the more I look at leader election code in context of multi-revision and multi-tenant, the more I'm worrying...

Could someone who understand this maybe write a small doc that we can discuss ? 
@howardjohn

Reply:
When we call shared informer factory's Start method, it will start all kinds of informers that are instantiated. 

Since the informer used by this controller is instantiated later than we start the factory, so call the factory's Start method again.

But with this pr, i donot think it is necessary now.  We can remove it.